internal/repository/targets_test.go

```diff
@@ -167,6 +167,87 @@ func TestUpdateDelegation(t *testing.T) {
 	})
 }
 
+func TestReorderDelegations(t *testing.T) {
+	targetsSigner, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(targetsKeyBytes) //nolint:staticcheck
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("all_cases", func(t *testing.T) {
+		r := createTestRepositoryWithPolicy(t, "")
+
+		ruleNames := []string{"rule-1", "rule-2", "rule-3"}
+		for _, ruleName := range ruleNames {
+			err := r.AddDelegation(testCtx, targetsSigner, policy.TargetsRoleName, ruleName, nil, nil, 1, false)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		state, err := policy.LoadCurrentState(testCtx, r.r, policy.PolicyStagingRef)
+		if err != nil {
+			t.Fatal(err)
+		}
+		targetsMetadata, err := state.GetTargetsMetadata(policy.TargetsRoleName)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		initialOrder := []string{}
+		for _, role := range targetsMetadata.Delegations.Roles {
+			initialOrder = append(initialOrder, role.Name)
+		}
+		expectedInitialOrder := append([]string{}, "protect-main")
+		expectedInitialOrder = append(expectedInitialOrder, ruleNames...)
+		expectedInitialOrder = append(expectedInitialOrder, "gittuf-allow-rule")
+		assert.Equal(t, expectedInitialOrder, initialOrder)
+
+		// Case 1: Valid Input
+		newOrder := []string{"rule-3", "rule-1", "rule-2", "protect-main"}
```

Comment:
Ah I see, the tests are more complete here. I think it makes more sense to test in the policy package, which is where we actually catch the different errors.

Reply:
It does make sense! Now, I kept only the valid case and move the complete test suite to the policy package